templates/tempest_run.sh.j2

```diff
@@ -2,34 +2,45 @@
-export http_proxy={{ proxy_env["http_proxy"] | default('') }}
+export http_proxy='{{ proxy_env["http_proxy"] | default('') }}'
 unset https_proxy
-{% if tempest_must_be_run_in_screen -%}
+{%- if tempest_must_be_run_in_screen -%}
 if [[ "$TERM" =~ "screen" ]]; then
-{% endif -%}
-  {% if rally_install_method == 'package' -%}
-  source /home/rally/rally/bin/activate
-  {% endif -%}
-  {{ rally_cmd }} deployment use {{ item.key }}
-  verifier_id=$({{ rally_cmd }} verify list-verifiers | grep {{ item.key }} | awk '{print($2)}')
-  if [ -z "${verifier_id}" ]; then
-    echo 'Verifier ID not found for {{ item.key }}'
-    exit 1
-  fi
-  {{ rally_cmd }} verify use-verifier --id "${verifier_id}"
-  {{ rally_cmd }} verify start --skip-list {{ deployments_folder }}/{{ item.key }}-skip-list.yml --concurrency {{ concurrency }} --detailed $*
-  {% if item.value.tempest_manual_forloop_cleanup %}
-  # Remove any resources with tempest in their names. Perhaps acceptable in a devel/test environment.
-  source /home/rally/.rally/openrc
-  for snapshot in $({{ openstack_cmd }}openstack volume snapshot list --all-projects -f csv -c ID -c Name |grep -i tempest-|tr -d '"'|cut -d "," -f1); do echo \#Deleting Tempest volume snapshot $snapshot; {{ openstack_cmd }}openstack volume snapshot delete $snapshot; done
-  for volume in $({{ openstack_cmd }}openstack volume list --all-projects -f csv -c ID -c Name |grep -i tempest-|tr -d '"'|cut -d "," -f1); do echo \#Deleting Tempest volume $volume; {{ openstack_cmd }}openstack volume set --state error $volume; openstack volume delete $volume; done
-  for image in $({{ openstack_cmd }}openstack image list -f csv -c ID -c Name |grep -i tempest-|tr -d '"'|cut -d "," -f1); do echo \#Deleting Tempest image $image; {{ openstack_cmd }}openstack image delete $image; done
-  for project in $({{ openstack_cmd }}openstack project list -f csv -c ID -c Name |grep -i tempest-|tr -d '"'|cut -d "," -f1); do echo \#Deleting Tempest project $project; {{ openstack_cmd }}openstack project delete $project;done
-  for domain in $({{ openstack_cmd }}openstack domain list -f csv -c ID -c Name|tr -d '"'|grep tempest-test_domain|cut -d "," -f1); do echo \#Disabling and deleting Tempest domain $domain; {{ openstack_cmd }}openstack domain set --disable $domain; openstack domain delete $domain; done
-  {% endif %}
-{% if tempest_must_be_run_in_screen -%}
+{%- endif -%}
+{%- if rally_install_method == 'package' -%}
+source /home/rally/rally/bin/activate
+{%- endif -%}
+
+{%- if item.value.tempest_manual_forloop_cleanup %}
+# Remove any resources with tempest in their names. Perhaps acceptable in a devel/test environment.
+source /home/rally/.rally/openrc
+for snapshot in $({{ openstack_cmd }}openstack volume snapshot list --all-projects -f csv -c ID -c Name |grep -i tempest-|tr -d '"'|cut -d "," -f1); do echo \#Deleting Tempest volume snapshot $snapshot; {{ openstack_cmd }}openstack volume snapshot delete $snapshot; done
```

Comment:
This can probably never run against production, it's quite risky listing everything and just grepping "tempest"

Reply:
Then we need to hard code what a production environment is. So, if someone else use this role, they have to adhere to our naming convention of what is prod what is not. I guess that was the intention of the tempest_manual_forloop_cleanup variable.